lib/model/model.go

```diff
@@ -2445,29 +2445,32 @@ func (m *model) GlobalDirectoryTree(folder, prefix string, levels int, dirsonly
 			return true
 		}
 
-		fullName := strings.Split(f.Name, sep)
-		name := fullName[len(fullName)-1]
-		path := fullName[:len(fullName)-1]
+		name := filepath.Base(f.Name)
+		path := filepath.Dir(f.Name)
 
-		tree := DirectoryTree{Name: name, IsDirectory: f.IsDirectory(), Children: []*DirectoryTree{}}
-		for !reflect.DeepEqual(path, currentPrefix) && len(currentPrefix) != 0 {
+		var children []*FolderItem = nil
+		if f.FileType() == 1 {
+			children = make([]*FolderItem, 0)
+		}
+		tree := FolderItem{Name: name, FileType: f.FileType(), Children: children}
+		for strings.Join(currentPrefix, sep) != path && len(currentPrefix) != 0 {
 			breadcrumbs = breadcrumbs[:len(breadcrumbs)-1]
 			currentPrefix = currentPrefix[:len(currentPrefix)-1]
 		}
 
-		if !dirsonly || f.IsDirectory() {
+		if !dirsonly || f.IsDirectory() && !f.IsIgnored() {
 			breadcrumbs[len(breadcrumbs)-1].Children = append(breadcrumbs[len(breadcrumbs)-1].Children, &tree)
 		}
 
-		if f.IsDirectory() {
+		if f.IsDirectory() && !f.IsIgnored() {
```

Comment:
Should f.IsIgnored() be a flag that is part of FolderItem which is visible but potentially greyed out in the tree view?
If not, perhaps this should an additional parameter on the rest API call? Seems like this will become a generally useful endpoint.